src/status_im/contexts/keycard/events.cljs

```diff
@@ -103,14 +66,51 @@
    {:fx [[:dispatch [:keycard/cancel-connection]]
          [:dispatch [:keycard/hide-connection-sheet]]]}))
 
+(rf/reg-event-fx :keycard/on-application-info-error
+ (fn [{:keys [db]} [error]]
+   {:db (assoc-in db [:keycard :application-info-error] error)
+    :fx [[:dispatch [:keycard/disconnect]]
+         [:dispatch
+          [:open-modal
+           (if (= :keycard/error.not-keycard error)
+             :screen/keycard.not-keycard
+             :screen/keycard.error)]]]}))
+
+(rf/reg-event-fx :keycard/update-application-info
+ (fn [{:keys [db]} [app-info]]
+   {:db (-> db
+            (assoc-in [:keycard :application-info] app-info)
+            (assoc-in [:keycard :application-info-error] nil))}))
```

Comment:
yeah but unfortunately there is no dissoc-in

Reply:
Yeah, people requested `dissoc-in` for a long time. We just use `update-in` with `dissoc` to remove keys.

Reply:
in most cases i don't see the difference in having nil or not having the key, but assoc-in is simpler and more readable, to you think we still should use dissoc for some reasons? thanks

Reply:
@flexsurfer, nil being used as absence of a key is fine most of the time, but not always. It's more precise to dissoc. For example, a developer may use `(when (contains? m k) ...)`, but if the nil value is stored, this will return true, which is not what the dev wanted probably. Some might say we shouldn't use `contains?` and only use `get` because it would cover for nil, like `(when (or (get m k)) ...)`. But IMO `contains?` is a nice function that communicates that I don't care about the return value, so I tend to use it whenever I can.

Reply:
сool. thank you i'll change it then